client/lib/plans/test/can-upgrade-to-plan.js

```diff
@@ -0,0 +1,116 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import { canUpgradeToPlan } from '..';
+import {
+	PLAN_BUSINESS,
+	PLAN_FREE,
+	PLAN_JETPACK_BUSINESS,
+	PLAN_JETPACK_BUSINESS_MONTHLY,
+	PLAN_JETPACK_FREE,
+	PLAN_JETPACK_PERSONAL,
+	PLAN_JETPACK_PERSONAL_MONTHLY,
+	PLAN_JETPACK_PREMIUM,
+	PLAN_JETPACK_PREMIUM_MONTHLY,
+	PLAN_PERSONAL,
+	PLAN_PREMIUM,
+} from '../constants';
+
+describe( 'canUpgradeToPlan', () => {
+	it( 'should return true from lower-tier plans to higher-tier plans', () => {
+		const makeSite = product_slug => ( {
+			plan: { product_slug },
+		} );
+
+		[
+			[ PLAN_FREE, PLAN_PERSONAL ],
+			[ PLAN_PERSONAL, PLAN_PREMIUM ],
+			[ PLAN_PREMIUM, PLAN_BUSINESS ],
+			[ PLAN_JETPACK_FREE, PLAN_JETPACK_PERSONAL ],
```

Comment:
Users are also able to upgrade from Jetpack free to any yearly Jetpack plan, not just Personal.